FactorioBlueprintStringRenderer/src/com/demod/fbsr/fp/FPRotatedSprite.java

```diff
@@ -123,6 +126,9 @@ private int getIndex(double orientation) {
 		if (backEqualsFront) {
 			index = index % (directionCount / 2);
 		}
+		if (index == directionCount) {
+			index = 0;
+		}
```

Comment:
I'm not sure this is the best fix, but the earlier math can result in index == directionCount which equals the list length, which leads to ArrayIndexOutOfBoundsException.

Reply:
good catch!